### ISSUE TITLE
TODO/SPIKE: eslint integration

### ISSUE THREAD

**Issue by hex13** [hex13/atom-lupa#37]
check if [linter-eslint package](https://atom.io/packages/linter-eslint) can be easy integrated with Atom-Lupa.

Need for some inter-package communication.

Related: #36, #31

**Comment by hex13** [comment]
I'm not sure if this is good direction.

ESLint is great and pluggable AST framework with great ecosystem, but most important thing from ESLint for this project is scopeManager which is basically EScope (other library ESLint only uses).

What does EScope do? It analyzes JS syntax tree and returns hierarchical structure with scopes and variables. 
http://mazurov.github.io/escope-demo/

I've used this in other project for implementing scope aware autocomplete, so it has power. 

Although. EScope  lacks of many analysis I need (support for objects e.g.:

```
const o = {
  abc: 3,
  def: 4
}
```

besides EScope objects are not serializable to JSON so I have visit each node and serialize them (trivial task and probably not very CPU consuming, but still... ).

So I would have to hack this a little further to change for my needs.

Returning to ESLint. Reusing AST alone from ESLint could be optimization gain, but I still don't think it should be done on Atom level. Maybe more proper way to do this would be reimplement backend of Lupa as a ESLint plugin. 

But complexity grows. 

On the other hand:
-  it turns out that gathering scoping of JavaScript with references etc. is pretty simple task, I made some proof of concept myself, I just have to polish that. So I think I just refactor or rewrite Lupa backend to handle scoping.
- having system specially designed for need of my projects could be a lot simpler that hack existing frameworks (which were designed for other purposes than general purpose code intelligence engine)

hex13/lupa#15
